### DISCUSSION TITLE
Are secrets working?

### DISCUSSION THREAD

**Discussion by davidemoraschi** [community/community#18081] [Codespaces] [answered]
I cannot build a codespace referencing a secret from the yml file

**Comment by craiglpeters** [comment]
Secrets are definitely working for most folks. Can you be a bit more specific about what is not working? I'm assuming you're using the [documentation on using secrets](https://docs.github.com/en/codespaces/managing-your-codespaces/managing-encrypted-secrets-for-your-codespaces).

**Reply by davidemoraschi** [reply]
Hi Craig, thanks for the answer. Yes, I do follow the documentation.
I have reproduced the error in this repo: https://github.com/davidemoraschi/codespaces-shared
If you don't mind, can you tell me what's wrong?

When I use in docker-compose.yml:
`POSTGRES_PASSWORD: ${{ secrets.POSTGRES_PASSWORD }}
`
the codespace does not build, when I use:
`POSTGRES_PASSWORD: postgres
`
the codespace builds correctly.

**Reply by davidemoraschi** [reply]
This is my secret:
![image](https://user-images.githubusercontent.com/9515147/172787309-098aa90c-e73e-4075-a7a1-739769d5c3e9.png)

**Reply by joshspicer** [reply]
Hey! We don't currently offer the "actions-esque" `${{ secrets.POSTGRES_PASSWORD }}` syntax at build time like you demonstrated above.  We currently expect users to use secrets after build time, like in the `postCreateCommand` lifecycle script and their application code.  Secrets are exposed as environment variables (eg: in bash script you could use `$POSTGRES_PASSWORD`, or in node `process.env.POSTGRES_PASSWORD`)

You have highlighted some documentation gaps that our docs team is going to take a stab at updating, so thank you for that :) 

Let me know if I can clarify further!

**Reply by joshspicer** [reply]
I am now seeing the workaround that @smajchrz shared below.  I'm not too familiar with this strategy, and is not something we directly support, but looks like it could work for your specific use-case.

**Comment by smajchrz** [comment, marked as answer]
This workaround allows the use of secrets before building a codespace container.
https://github.com/microsoft/vscode-remote-release/issues/4841#issuecomment-1137380363

**Reply by davidemoraschi** [reply]
Thanks @smajchrz, good point: I'll use the workaround proposed in:
https://github.com/microsoft/vscode-remote-release/issues/4841#issuecomment-815107551
@joshspicer sure, having the "actions-esque" syntax would be very good, and consistent with GitHub action
